testsuite/forge/src/success_criteria.rs

```diff
@@ -26,13 +35,15 @@ impl SuccessCriteria {
         check_no_restarts: bool,
         wait_for_all_nodes_to_catchup: Option<Duration>,
         system_metrics_threshold: Option<SystemMetricsThreshold>,
+        chain_progress_check: Option<StateProgressThreshold>,
```

Comment:
Does it make sense to do progress check and wait for catchup? If not, should we add an assert here?

Reply:
@bchocho :

- progress check makes sure that chain itself doesn't have any large pauses. if less than f+1 nodes are fully down, it can pass
- wait for catchup doesn't have any checks on how consistently the chain/individual nodes progressed, but makes sure at the end that there are no stragglers. so if there is one node down, unable to catchup , that check will fail.


so they are orthogonal/complimentary check.

Reply:
but if catchup is not set when progress check is, I need to actually pick the client that has the most recent version, as if I pick the down node, chain_progress_check is not checking what it should. I'll make that change